docs/OPENSHIFT.md

```diff
@@ -0,0 +1,184 @@
+
+By default the agent is configured to run as root; however, the DaemonSet can be modified to run the agent as a non-root user.
+
+This is accomplished through Linux capabilities and turning the agent binary into a "capability-dumb binary." The binary is given `CAP_DAC_READ_SEARCH` to read all files on the file system. The image already comes with this change and the necessary user and group. The only required step is configuring the agent DaemonSet to run as the user and group `5000:5000`.
+
+Add two new fields, `runAsUser` and `runAsGroup`, to the `securityContext` section found in the `logdna-agent` container in the `logdna-agent` DaemonSet inside of `k8s/agent-resources-openshift.yaml` [`spec.template.spec.containers.0.securityContext`]:
+
+```yaml
+securityContext:
+  runAsUser: 5000
+  runAsGroup: 5000
+```
+
+Apply the updated configuration to your cluster:
+
+```console
+oc apply -f k8s/agent-resources-openshift.yaml
+```
+
+Alternatively, update the DaemonSet configuration directly by using the following patch command:
+
+```console
+oc patch daemonset logdna-agent --type json -p '[{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsUser","value":5000},{"op":"add","path":"/spec/template/spec/containers/0/securityContext/runAsGroup","value":5000}]'
+```
+
+## Collecting Node Journald Logs
+
+The agent by default only captures logs generated by the containers running on the OpenShift cluster's container runtime environment. It does not, however, collect system component logs from applications running directly on the node such as the kubelet and container runtime. With some configuration on both the node and the agent, these journald logs can be exposed from the node to the agent.
+
+The agent can access Journald logs from the host node by mounting the logs from `/var/log/journal`. This requires enabling journald log storage in the node as well as configuring the agent to monitor the directory.
+
+### Enabling Journald on the Node
+
+Follow OpenShifts documentation for [enabling Journald on your nodes](https://docs.openshift.com/container-platform/4.6/logging/config/cluster-logging-systemd.html).
+
+### Enabling Journald Monitoring on the Agent
+
+To enable Journald monitoring in the agent, add a new environment variable, `LOGDNA_JOURNALD_PATHS` with a value of `/var/log/journal`, to the logdna-agent DaemonSet:
+* If you are updating an already deployed agent:
+  1. You can patch the existing agent by running.
+```console
+oc patch daemonset -n logdna-agent logdna-agent --type json -p '[{"op":"add","path":"/spec/template/spec/containers/0/env/-","value":{"name":"LOGDNA_JOURNALD_PATHS","value":"/var/log/journal"}}]'
+```
+* If you are modifying a YAML file:
+  1. Add the new environment variable to the envs section of the DaemonSet object in `k8s/agent-resources-openshift.yaml` [`spec.template.spec.containers.0.env`].
+  2. Apply the new configuration file, run `oc apply -f k8s/agent-resources-openshift.yaml`.
```

Comment:
177 and 178 beta yaml @c-nixon

Reply:
These ones refer to github, so they should be ok, I think it's only the cdn ones that we'll have issues with